apps/08jquery-towers-of-hanoi/script.js

```diff
@@ -1,5 +1,41 @@
 'use strict';
 
+var block = null;
+
 $(document).ready(function() {
-  // Put app logic here
+  
+  $('[data-stack]').click(dataStackWasClicked);// this is the click listener
+
+  function dataStackWasClicked(){
+
+    if(block){// This creates the variables needed to make a vailid move
+      var lastBlockOfTheStack = $(this).children().last();//Creates a variable for the last block on a stack
+      var dataBlockAttributeoflastblock = lastBlockOfTheStack.data("block");//this creates a variable for the data attributes of the last block
+      var dataAttributeOfRemovedBlock = block.data("block");//this vreates a variable for the data attributes of the removed block
+
+      if(dataBlockAttributeoflastblock > dataAttributeOfRemovedBlock
+      || lastBlockOfTheStack.length === 0){//this compares the data attributes of the last block and the removed block to determin if the move is valid
+        $(this).append(block);//if the move is valid, add removed block to the stack
+        block = null;//then return block to null
+      }
+    }
+    else{
+      block = $(this).children().last().detach();//otherwise block still holds the last block clicked.
```

Comment:
If you declare `var lastBlockOfTheStack` (line 12) outside your first if statement, you could reuse the logic here. 

`ex: block = lastBlockOfTheStack.detach();`